### DISCUSSION TITLE
Public Access Network screen ideas

### DISCUSSION THREAD

**Discussion by mefistotelis** [swfans/syndwarsfx#170] [General]
Any ideas of what exactly the PAN screen was supposed to contain?

A list of extra missions to select from?
Something similar to a chat which enables the additional objectives upon clicking through a conversation?
Just a progress of additional objectives, which would come up by themselves during regular missions?

![synII183](https://github.com/swfans/swars/assets/11463396/c7959b1b-3b33-4374-9502-ad3828a55708)

The screen was either commented out from final code or never implemented at all (most likely the latter) so this is all there is now.

**Comment by Moburma** [comment]
I've thought about this before and could never really make head nor tail of it, Sean Masterson's explanations only made it more confusing!
The one thing we do know is that it would seemingly have new "items" that would get unlocked via level completion based on the .miss data structure. 
I imagine it as something like a mix between the mission briefings and some kind of in-game BBS system.

I wonder if it at one point you would only get Unguided missions/objectives/whatever here, then they moved towards the less ambitious idea of simply having an unguided campaign, and finally just scrapping the unguided entirely.

I definitely agree it's unlikely they ever actually implemented anything, just seems to be placeholder stuff.

I wonder/kind of assume the initial idea was more that by using this you would essentially be working as a double agent yourself against Eurocorp/Zealots by participating in this and aiding the unguided. I have no idea how that would work in the bigger picture though.

**Comment by mefistotelis** [comment]
A BBS system? That makes me remember a part of code for opening new mail:

```
        if (open_brief < 0) {
            short email;
            email = -open_brief - 1;
            sprintf(fname, "%s/mail%03d.txt", "textdata", email_store[email].Mission);
        } else if (open_brief > 0) {
            ushort missi;
            missi = brief_store[open_brief - 1].Mission;
            sprintf(fname, "%s/miss%03d.txt", "textdata", mission_list[missi].SourceID);
        }
```

So in the final game, `open_brief` is always positive so only `missNNN.txt` are used to show the brief. Maybe `mailNNN.txt` were supposed to be the additional messages from the BBS?

So were the BBS messages supposed to be delivered under "new mail" icon like normal missions? But then BBS screen wouldn't be required.. maybe after interacting with the BBS through PAN screen you would get the email with mission (or alternate objective) details?

**Reply by mefistotelis** [reply]
That would kind of make sense:
* By completing missions, you enable a range of PAN BBS messages - some are for lore only, some are suggesting some kind of call to action in a specific city.
* In case of CTA messages, you can contact the author and they will email you the mission details.
* The email you receive informs about additional objective in a specific city; you cannot start mission in the city based on this email, but you can accomplish the objective while doing the normal Eurocorp missions.

**Reply by Moburma** [reply]
Hmm, that is interesting. I always thought that as it was a _Public_ Access Network it would be basically kind of a public facing system like usenet, where everyone can post. It would be somewhat anarchic, but obviously the corps would be privvy and could stop things, so it would be a kind of cat and mouse game of the unguided (Vissick) trying to get as many eyes on a message as possible without incriminating themselves or revealing their plans too early.

EDIT:  Yeah, your comment above nails it!

**Reply by Moburma** [reply]
The other missing piece of the puzzle is the in-game element. Masterson talks about there being actual PAN consoles in levels, and about how going near them (and suggesting NPC unguided going near them would also trigger it) would unlock more items or something in the screen. Then there are of course actual leftover objectives to specifically use the PAN in-game. What's odd is that there really isn't any thing that looks like a computer console to me in-game, maybe those things that are used in the orbital station level? Again suggests the idea didn't go far at all.

**Reply by mefistotelis** [reply]
Yeah, that's hard to put into a working mechanic..

I would understand if the email received after BBS would say "We will provide detailed instructions on PAN terminal near guns shop" and then you need to reach the terminal (yes, I think the orbital station / Vancouver terminals are the ones) to get your objective updated.

But getting NPC Unguided there? I assume not a persuaded one, so then the NPC would have a command to move there, you would only have to trigger it? And then what, after he reaches the terminal he goes back to you to share the new information? Doesn't sound very fun, nor very convincing.

Maybe the Unguided NPC thread is an additional idea built over the original mechanics. Like, normally you are the one who gets to the terminals, but in one level you see Unguided NPC doing the same - performing a PAN mission, like you did before. Through this mission, he may open a dome or something, allowing access to a new item. But he will fail without your intervention.
That would make sense.

Btw, in one level I've seen a dome with Graviton Gun inside.. I think it was one of the MP maps? No idea how player should open it, but destroying the dome destroyed the gun.

**Reply by Moburma** [reply]
Ahh, I never noticed that dome before. Looking into it, there actually IS an invisible NPC that opens the dome when you walk near it, but it has a weird criteria that an object needs destroying first (other dome?). I will make the person non-invisible and have a look at what is going on, it's another tricksy thing like the zealot outside the church on the Hong Kong level, using UNTIL again.

That is something I thought of suggesting as well - an enhancement so that domes have no collision when fully opened. It's fairly clear they just don't work properly in the final game and were abandoned as a device. If they could actually be entered when opened then all kinds of interesting things could be built around it, like hiding items in there, or vehicles. The scripted sequence with the soldier in the dome in the mission The Dead Zone would actually look right then.
I actually wonder why they didn't get this working, they're clearly an old game element that goes back to the pre-alpha, but ended up useless - unless they _did_ get it working but it caused other issues?

**Comment by mefistotelis** [comment]
> they're clearly an old game element that goes back to the pre-alpha, but ended up useless - unless they did get it working but it caused other issues?

Re-computing the routing triangulation of an area based on 3D geometry intersecting the ground is kind of taxing in regard to CPU cycles. There was a noticeable slowdown when a building ends collapsing and the triangulation below switches flags to be passable, and that part was especially optimized.

So I guess that was the issue - the pathfinding in pre-alpha was simpler and could handle that in real-time, but when they made 'proper' pathfinding that become problematic.

For opening the dome, simplified procedure could be used- as for collapsing.
But then closing it, would require expensive re-computation, or a different kind of smart optimization.
Shouldn't be that much of a problem today though.

**Reply by Moburma** [reply]
Figured out the above:

![](https://i.imgur.com/dsr57MW.png)

So you just have to go into the little building and it opens the dome. Once again deliberate weird stuff with the UNTIL command - maybe that was to stop it opening if destroyed? It makes no difference here and just works.

Of course the problem it's impossible to get the gun because it's stuck in the dome!

I hadn't considered the problem of pathfinding, that makes a lot of sense why such a feature would be disabled. Things like this suggests it DID work at one point though. 

EDIT: 
Also it looks like they started something similar probably for the other dome. There is another invisible NPC who is oddly by the raised square building on the other side of the river. He only has one command, to run to a point nearby. However, the next command along is a WAIT_MEM_G_ARRIVE with coordinates very near the second dome. There are no other commands and the group arrive command is not even hooked up to his first command.

**Reply by mefistotelis** [reply]
Right, the mission is `-m 99,16` and the dome does open as long as one of your agents is in the building.
